Terminal.Gui/Core/Application.cs

```diff
@@ -114,6 +114,19 @@ public static class Application {
 		/// <summary>
 		/// The current <see cref="ConsoleDriver.HeightAsBuffer"/> used in the terminal.
 		/// </summary>
+		/// <remarks>
+		/// <para>
+		/// If <see langword="false"/> (the default) the height of the Terminal.Gui application (<see cref="ConsoleDriver.Rows"/>) 
+		/// tracks to the height of the visible console view when the console is resized. In this case 
+		/// scrolling in the console will be disabled and all <see cref="ConsoleDriver.Rows"/> will remain visible.
+		/// </para>
+		/// <para>
+		/// If <see langword="true"/> then height of the Terminal.Gui application <see cref="ConsoleDriver.Rows"/> does not
+		/// change when the console is resized on shrinking (it only grows and never decrease). 
+		/// In this case console scrolling is enabled and the contents (<see cref="ConsoleDriver.Rows"/> high) will scroll
+		/// as the console scrolls. 
+		/// </para>
+		/// </remarks>
```

Comment:
I think we should rename this property "EnableConsoleScrolling". Do it like this:

- Add a new `public bool EnableConsoleScrolling { get; set; }`
- Use `EnableConsoleScrolling` throughout the existing codebase (including UI Catalog).
- Keep `HeightAsBuffer` as a public, having it just get/set `EnableConsoleScrolling`. 
- Make the API Docs for `HeightAsBuffer` read "This API is deprecated; use `EnableConsoleScrolling` instead.".
- Add a remark <para> to the API docs for `EnableConsoleScrolling` saying "This API was previously named 'HeightAsBuffer` but was renamed to make its purpose more clear."